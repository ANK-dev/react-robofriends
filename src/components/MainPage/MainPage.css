@import '../../fonts/SEGA.css';

body {
    margin: 0;
    padding: 0;
    font-family: sans-serif;
    background: linear-gradient(to left, rgba(7,27,82,1) 0%, rgba(0,128,128,1) 100%);
    max-height: 50%;
}

h1 {
    font-family: 'SEGA LOGO FONT';
    font-weight: 200;
    color: #0ccac4;
}