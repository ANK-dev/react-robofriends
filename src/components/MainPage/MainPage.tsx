
import React, { useEffect } from 'react';

import CardList from '../../components/Cards/CardList';
import SearchBox from '../../components/SearchBox/SearchBox';
import Scroll from '../../components/Scroll/Scroll'
import ErrorBoundry from '../../components/ErrorBoundry/ErrorBoundry';
import Header from '../../components/Header/Header';

import './MainPage.css'

export interface IRobot {
  name: string;
  id: number;
  email: string;
}

function MainPage(props: any) {
    const {
        searchfield,
        onSearchChange,
        onRequestRobots,
        robots,
        isPending
    } = props;

    useEffect(() => {
            onRequestRobots()
        }, [])

    const filteredRobots: IRobot[] = robots.filter((robot: any) => {
        return robot.name.toLowerCase().includes(
            searchfield.toLowerCase()
        );
    })

    if (isPending)
        return (
            <div className='tc'>
                <Header/>
                <h1>Loading...</h1>
            </div>
        );
    else
        return (
            <div className='tc'>
                <Header/>
                <SearchBox searchChange={onSearchChange}/>
                <Scroll>
                    <ErrorBoundry>
                        <CardList robots={filteredRobots}/>
                    </ErrorBoundry>
                </Scroll>
            </div>
        );

}

export default MainPage;