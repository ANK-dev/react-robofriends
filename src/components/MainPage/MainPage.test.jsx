import MainPage from './MainPage';
import React from 'react';
import {shallow} from 'enzyme';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchfield: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps}/>);
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})