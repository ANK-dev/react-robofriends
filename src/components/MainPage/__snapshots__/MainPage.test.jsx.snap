// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <Cardlist
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
